File: backend/shophandmade/src/main/java/com/project/shophandmade/models/Chitietdonhang.java
package com.project.shophandmade.models;

import com.fasterxml.jackson.annotation.JsonBackReference;
import jakarta.persistence.*;
import lombok.*;

@Entity
@Table(name = "chitietdonhang")
@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@Builder
public class Chitietdonhang {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long Ma;

    @ManyToOne
    @JoinColumn(name = "Ma_Donhang")
    @JsonBackReference
    private Donhang donhang;

    @ManyToOne
    @JoinColumn(name = "Ma_Sanpham")
    private Sanpham sanpham;

    @Column(name = "Gia", nullable = false)
    private Float Gia;

    @Column(name = "Soluong", nullable = false)
    private int Soluong;

    @Column(name = "Thanhtien", nullable = false)
    private Float Thanhtien;

   

}

File: Frontend/shophandmade/src/app/models/Chitietdonhang.ts
import { Sanpham } from "./Sanpham";
import { Donhang} from "./Donhang";




export interface  Chitietdonhang {
    ma: number;
    donghang:Donhang ;
    sanpham: Sanpham;
    gia: number;
    soluong: number;
    thanhtien: number;
   
}